helm-chart/azp-agent/templates/azp-agent.yaml

```diff
@@ -17,15 +17,26 @@ spec:
       labels:
         app: {{ .Chart.Name }}
     spec:
+      {{- if .nodeSelector }}
+      nodeSelector:
+      {{ toYaml .nodeSelector | indent 2 }}
+      {{ end }}
       containers:
       - name: {{ .Chart.Name }}
         image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
         imagePullPolicy: {{ .Values.image.pullPolicy }}
+        {{- if .Values.resources.requests }}
+        resources:
+          requests:
+          {{ toYaml .Values.resources.requests | indent 2 }}
+          limits:
+          {{ toYaml .Values.resources.limits | indent 2 }}
+        {{ end }}
```

Comment:
Should we check if either `requests` or `limits` appears? I'm not sure if that's a valid use case. I.e. when people define limits do they necessarily define requests? 

I opted to keep the logic simple for now, but I'd like your feedback.

Reply:
So instead of this a common approach seems to be to just grab the entire resources object from Values. Something like:
```
        resources:
{{ toYaml .Values.server.resources | indent 10 }}
```

---
In reply to: [381463010](https://github.com/julioct/azure-pipelines-kubernetes-agents/pull/1#discussion_r381463010) [](ancestors = 381463010)

Reply:
Problem is, resources should be optional. If not defined, we don't want to specify any default values. 

`resources` also includes the disk options which need to not be copied here.

Reply:
Yea that's ok. You can add this to values.yaml:

``` resources: {}

Then you add this to this file (check the indent):
``` resources:
      {{- toYaml .Values.resources | nindent 12 }}

It should work just fine. If in doubt try creating a brand new char with the ``` helm create ``` command and see how it lays out the defaults for resources.

---
In reply to: [381616831](https://github.com/julioct/azure-pipelines-kubernetes-agents/pull/1#discussion_r381616831) [](ancestors = 381616831)